alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/MessageContentCollector.java

```diff
@@ -191,45 +205,41 @@ private JsonFieldAccessor createJsonAccessor(final List<JsonField<?>> notificati
         return jsonExtractor.createJsonFieldAccessor(notificationFields, notificationJson);
     }
 
-    private List<AggregateMessageContent> getContentsOrCreateIfDoesNotExist(final JsonFieldAccessor accessor, final List<JsonField<?>> notificationFields) {
-        final List<AggregateMessageContent> aggregateMessageContentsForNotifications = new ArrayList<>();
+    private List<ProviderMessageContent.Builder> getMessageBuildersOrCreateIfTheyDoNotExist(JsonFieldAccessor accessor, List<JsonField<?>> notificationFields) {
+        final List<ProviderMessageContent.Builder> buildersForNotifications = new ArrayList<>();
 
         final List<LinkableItem> topicItems = getTopicItems(accessor, notificationFields);
         if (topicItems.isEmpty()) {
             return List.of();
         }
         final List<LinkableItem> subTopicItems = getSubTopicItems(accessor, notificationFields);
-        // for the number of topics assume there is an equal number of sub topics and the order is the same.this seems fragile at the moment.
+        // For the number of topics assume there is an equal number of sub topics and the order is the same. This seems fragile at the moment.
         final int count = topicItems.size();
         for (int index = 0; index < count; index++) {
             final LinkableItem topicItem = topicItems.get(index);
 
-            final Optional<LinkableItem> subTopic;
+            LinkableItem subTopic = null;
             if (!subTopicItems.isEmpty()) {
-                subTopic = Optional.ofNullable(subTopicItems.get(index));
-            } else {
-                subTopic = Optional.empty();
+                subTopic = subTopicItems.get(index);
             }
 
-            final String subTopicName = subTopic.map(LinkableItem::getName).orElse(null);
-            final String subTopicValue = subTopic.map(LinkableItem::getValue).orElse(null);
-            final AggregateMessageContent foundContent = findTopicContent(topicItem.getName(), topicItem.getValue(), subTopicName, subTopicValue);
+            final LinkableItem providerItem = getProviderItem();
+            final ProviderMessageContent.Builder foundContent = findContentBuilder(providerItem.getValue(), topicItem, subTopic);
 
-            if (foundContent != null) {
-                aggregateMessageContentsForNotifications.add(foundContent);
+            if (null != foundContent) {
+                buildersForNotifications.add(foundContent);
             } else {
-                final SortedSet<CategoryItem> categoryItems = new TreeSet<>();
-                aggregateMessageContentsForNotifications.add(createAggregateMessageContent(topicItem, subTopic, categoryItems));
+                ProviderMessageContent.Builder builder = new ProviderMessageContent.Builder();
+                builder
+                    .applyProvider(providerItem.getValue(), providerItem.getUrl().orElse(null))
```

Comment:
Can this line be used on line 233 instead and you add these items immediately after creating the builder? The formatting is weird here with a single item on 233